rs/sns/governance/src/extensions.rs

```diff
@@ -1279,10 +1399,148 @@ mod tests {
+
+        let root_canister_id = CanisterId::from_u64(1001);
+        let governance_proto = GovernanceProto {
+            root_canister_id: Some(root_canister_id.get()),
+            ledger_canister_id: Some(CanisterId::from_u64(4000).get()),
+            swap_canister_id: Some(CanisterId::from_u64(5000).get()),
+            parameters: Some(NervousSystemParameters::with_default_values()),
+            sns_metadata: Some(SnsMetadata {
+                logo: None,
+                url: Some("https://example.com".to_string()),
+                name: Some("Test SNS".to_string()),
+                description: Some("Test SNS for extensions".to_string()),
+            }),
+            mode: governance::Mode::Normal.into(),
+            ..Default::default()
+        };
+
+        // Create mocks with configured expectations
+        let mut sns_ledger = MockICRC1Ledger::new();
+        let mut icp_ledger = MockICRC1Ledger::new();
+
+        // Get the expected subaccounts
+        let governance_canister_id = env.canister_id();
+
+        Some(compute_distribution_subaccount_bytes(
+            governance_canister_id.get(),
+            TREASURY_SUBACCOUNT_NONCE,
+        ));
+
+        let sns_subaccount = compute_distribution_subaccount_bytes(governance_canister_id.get(), 0);
+
+        // Configure SNS ledger mock
+        sns_ledger
+            .expect_account_balance()
+            .withf(move |account: &Account| {
+                account.owner == governance_canister_id.get().0
+                    && account.subaccount == Some(sns_subaccount)
+            })
+            .times(1)
+            .returning(move |_| Ok(Tokens::from_e8s(sns_balance)));
+
+        // Configure ICP ledger mock
+        icp_ledger
+            .expect_account_balance()
+            .withf(move |account: &Account| {
+                account.owner == governance_canister_id.get().0 && account.subaccount == None
+            })
+            .times(1)
+            .returning(move |_| Ok(Tokens::from_e8s(icp_balance)));
+
+        Governance::new(
+            ValidGovernanceProto::try_from(governance_proto)
+                .expect("Failed validating governance proto"),
+            Box::new(env),
+            Box::new(sns_ledger),
+            Box::new(icp_ledger),
+            Box::new(MockCMC::default()),
+        )
+    }
+
+    #[tokio::test]
+    async fn test_validate_deposit_operation_treasury_balance_limits() {
+        // Test parameters: (sns_balance, icp_balance, sns_request, icp_request, should_succeed, error_msg)
+        let test_cases = vec![
+            // Positive case: both amounts at exactly 50%
+            (100_000_000, 200_000_000, 50_000_000, 100_000_000, true, ""),
+            // Positive case: both amounts below 50%
+            (100_000_000, 200_000_000, 30_000_000, 60_000_000, true, ""),
+            // Positive case: zero amounts
+            (100_000_000, 200_000_000, 0, 0, true, ""),
```

Comment:
These nice comments could be turned into labels that are printed in case the assertion in the loop body fails (this speeds up finding which case to debug).

```suggestion
            ("Positive case: zero amounts", 100_000_000, 200_000_000, 0, 0, true, ""),
```